packages/block-editor/src/components/rich-text/event-listeners/delete.js

```diff
@@ -30,8 +30,8 @@ export default ( props ) => ( element ) => {
 				return;
 			}
 
-			// Exclude shift+backspace as they are shortcuts for deleting blocks.
-			if ( shiftKey ) {
+			// Exclude (command|ctrl)+shift+backspace as they are shortcuts for deleting blocks.
+			if ( ( ctrlKey && shiftKey ) || ( metaKey && shiftKey ) ) {
```

Comment:
```suggestion
			if ( shiftKey && ( ctrlKey || metaKey ) ) {
```

Let's simplify the conditions.